src/sql/workbench/browser/designer/designerIssuesTabPanelView.ts

```diff
@@ -0,0 +1,146 @@
+			mouseSupport: true,
+			accessibilityProvider: new DesignerIssueListAccessibilityProvider()
+		});
+		this._register(this._issueList.onDidChangeSelection((e) => {
+			if (e.elements && e.elements.length === 1) {
+				this._onIssueSelected.fire(e.elements[0].propertyPath);
+			}
+		}));
+		this._register(attachListStyler(this._issueList, this._themeService));
+	}
+
+	layout(dimension: DOM.Dimension): void {
+		this._issueList.layout(dimension.height, dimension.width);
+	}
+
+	updateIssues(errors: DesignerIssue[]) {
+		if (this._issueList) {
+			this._issueList.splice(0, this._issueList.length, errors);
+		}
+	}
+}
+
+registerThemingParticipant((theme: IColorTheme, collector: ICssStyleCollector) => {
+	const errorForegroundColor = theme.getColor(problemsErrorIconForeground);
+	const warningForegroundColor = theme.getColor(problemsWarningIconForeground);
+	const informationalForegroundColor = theme.getColor(problemsInfoIconForeground);
+	if (errorForegroundColor) {
+		collector.addRule(`
+		.designer-component .issues-container .issue-item .issue-icon.codicon-error {
+			color: ${errorForegroundColor};
+		}
+		.designer-component .issues-container .issue-item .issue-icon.codicon-warning {
+			color: ${warningForegroundColor};
+		}
+		.designer-component .issues-container .issue-item .issue-icon.codicon-info {
+			color: ${informationalForegroundColor};
+		}
+		`);
+	}
+});
+
+const DesignerIssueListTemplateId = 'DesignerIssueListTemplate';
+class DesignerIssueListDelegate implements IListVirtualDelegate<DesignerIssue> {
+	getHeight(element: DesignerIssue): number {
+		return 25;
+	}
+
+	getTemplateId(element: DesignerIssue): string {
+		return DesignerIssueListTemplateId;
+	}
+}
+
+interface DesignerIssueListItemTemplate {
+	issueText: HTMLDivElement;
+	issueIcon: HTMLDivElement;
+}
+
+class TableFilterListRenderer implements IListRenderer<DesignerIssue, DesignerIssueListItemTemplate> {
+	renderTemplate(container: HTMLElement): DesignerIssueListItemTemplate {
+		const data: DesignerIssueListItemTemplate = Object.create(null);
+		const issueItem = container.appendChild(DOM.$('.issue-item'));
+		data.issueIcon = issueItem.appendChild(DOM.$(''));
+		data.issueText = issueItem.appendChild(DOM.$('.issue-text'));
+		return data;
+	}
+
+	renderElement(element: DesignerIssue, index: number, templateData: DesignerIssueListItemTemplate, height: number): void {
+		templateData.issueText.innerText = element.description;
+		templateData.issueText.title = element.description;
+		let iconClass;
+		switch (element.severity) {
+			case 'warning':
+				iconClass = Codicon.warning.classNames;
+				break;
+			case 'informational':
+				iconClass = Codicon.info.classNames;
+				break;
+			default:
+				iconClass = Codicon.error.classNames;
+				break;
+		}
+		templateData.issueIcon.className = `issue-icon ${iconClass}`;
+	}
+
+	disposeElement?(element: DesignerIssue, index: number, templateData: DesignerIssueListItemTemplate, height: number): void {
```

Comment:
Visibility modifier - and why is this optional here? That shouldn't be necessary in the implementation.

Reply:
removed.